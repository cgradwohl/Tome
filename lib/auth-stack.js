"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const aws_apigateway_1 = require("@aws-cdk/aws-apigateway");
const aws_lambda_1 = require("@aws-cdk/aws-lambda");
const core_1 = require("@aws-cdk/core");
const aws_cognito_1 = require("@aws-cdk/aws-cognito");
class AuthStack extends core_1.Stack {
    constructor(app, id) {
        super(app, id);
        // Function that returns 201 with "Hello world!"
        const helloWorldFunction = new aws_lambda_1.Function(this, 'helloWorldFunction', {
            code: new aws_lambda_1.AssetCode('lambdas'),
            handler: 'helloworld.handler',
            runtime: aws_lambda_1.Runtime.NODEJS_12_X
        });
        // Rest API backed by the helloWorldFunction
        const helloWorldLambdaRestApi = new aws_apigateway_1.LambdaRestApi(this, 'helloWorldLambdaRestApi', {
            restApiName: 'Hello World API',
            handler: helloWorldFunction,
            proxy: false,
        });
        // Cognito User Pool with Email Sign-in Type.
        const userPool = new aws_cognito_1.UserPool(this, 'userPool', {
            signInAliases: {
                email: true
            }
        });
        // Authorizer for the Hello World API that uses the
        // Cognito User pool to Authorize users.
        const authorizer = new aws_apigateway_1.CfnAuthorizer(this, 'cfnAuth', {
            restApiId: helloWorldLambdaRestApi.restApiId,
            name: 'HelloWorldAPIAuthorizer',
            type: 'COGNITO_USER_POOLS',
            identitySource: 'method.request.header.Authorization',
            providerArns: [userPool.userPoolArn],
        });
        // Hello Resource API for the REST API. 
        const hello = helloWorldLambdaRestApi.root.addResource('HELLO');
        // GET method for the HELLO API resource. It uses Cognito for
        // authorization and the auathorizer defined above.
        hello.addMethod('GET', new aws_apigateway_1.LambdaIntegration(helloWorldFunction), {
            authorizationType: aws_apigateway_1.AuthorizationType.COGNITO,
            authorizer: {
                authorizerId: authorizer.ref
            }
        });
    }
}
exports.AuthStack = AuthStack;
const app = new core_1.App();
new AuthStack(app, 'AuthStack');
app.synth();
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYXV0aC1zdGFjay5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImF1dGgtc3RhY2sudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSw0REFBNkc7QUFDN0csb0RBQW1FO0FBQ25FLHdDQUEyQztBQUMzQyxzREFBK0M7QUFFL0MsTUFBYSxTQUFVLFNBQVEsWUFBSztJQUNsQyxZQUFZLEdBQVEsRUFBRSxFQUFVO1FBQzlCLEtBQUssQ0FBQyxHQUFHLEVBQUUsRUFBRSxDQUFDLENBQUM7UUFFZixnREFBZ0Q7UUFDaEQsTUFBTSxrQkFBa0IsR0FBRyxJQUFJLHFCQUFRLENBQUMsSUFBSSxFQUFFLG9CQUFvQixFQUFFO1lBQ2xFLElBQUksRUFBRSxJQUFJLHNCQUFTLENBQUMsU0FBUyxDQUFDO1lBQzlCLE9BQU8sRUFBRSxvQkFBb0I7WUFDN0IsT0FBTyxFQUFFLG9CQUFPLENBQUMsV0FBVztTQUM3QixDQUFDLENBQUM7UUFFSCw0Q0FBNEM7UUFDNUMsTUFBTSx1QkFBdUIsR0FBRyxJQUFJLDhCQUFhLENBQUMsSUFBSSxFQUFFLHlCQUF5QixFQUFFO1lBQ2pGLFdBQVcsRUFBRSxpQkFBaUI7WUFDOUIsT0FBTyxFQUFFLGtCQUFrQjtZQUMzQixLQUFLLEVBQUUsS0FBSztTQUNiLENBQUMsQ0FBQztRQUVILDZDQUE2QztRQUM3QyxNQUFNLFFBQVEsR0FBRyxJQUFJLHNCQUFRLENBQUMsSUFBSSxFQUFFLFVBQVUsRUFBRTtZQUM5QyxhQUFhLEVBQUU7Z0JBQ2IsS0FBSyxFQUFFLElBQUk7YUFDWjtTQUNGLENBQUMsQ0FBQTtRQUVGLG1EQUFtRDtRQUNuRCx3Q0FBd0M7UUFDeEMsTUFBTSxVQUFVLEdBQUcsSUFBSSw4QkFBYSxDQUFDLElBQUksRUFBRSxTQUFTLEVBQUU7WUFDcEQsU0FBUyxFQUFFLHVCQUF1QixDQUFDLFNBQVM7WUFDNUMsSUFBSSxFQUFFLHlCQUF5QjtZQUMvQixJQUFJLEVBQUUsb0JBQW9CO1lBQzFCLGNBQWMsRUFBRSxxQ0FBcUM7WUFDckQsWUFBWSxFQUFFLENBQUMsUUFBUSxDQUFDLFdBQVcsQ0FBQztTQUNyQyxDQUFDLENBQUE7UUFFRix3Q0FBd0M7UUFDeEMsTUFBTSxLQUFLLEdBQUcsdUJBQXVCLENBQUMsSUFBSSxDQUFDLFdBQVcsQ0FBQyxPQUFPLENBQUMsQ0FBQztRQUVoRSw2REFBNkQ7UUFDN0QsbURBQW1EO1FBQ25ELEtBQUssQ0FBQyxTQUFTLENBQUMsS0FBSyxFQUFFLElBQUksa0NBQWlCLENBQUMsa0JBQWtCLENBQUMsRUFBRTtZQUNoRSxpQkFBaUIsRUFBRSxrQ0FBaUIsQ0FBQyxPQUFPO1lBQzVDLFVBQVUsRUFBRTtnQkFDVixZQUFZLEVBQUUsVUFBVSxDQUFDLEdBQUc7YUFDN0I7U0FFRixDQUFDLENBQUE7SUFFSixDQUFDO0NBQ0Y7QUFqREQsOEJBaURDO0FBR0QsTUFBTSxHQUFHLEdBQUcsSUFBSSxVQUFHLEVBQUUsQ0FBQztBQUN0QixJQUFJLFNBQVMsQ0FBQyxHQUFHLEVBQUUsV0FBVyxDQUFDLENBQUM7QUFDaEMsR0FBRyxDQUFDLEtBQUssRUFBRSxDQUFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgTGFtYmRhUmVzdEFwaSwgQ2ZuQXV0aG9yaXplciwgTGFtYmRhSW50ZWdyYXRpb24sIEF1dGhvcml6YXRpb25UeXBlIH0gZnJvbSAnQGF3cy1jZGsvYXdzLWFwaWdhdGV3YXknO1xuaW1wb3J0IHsgQXNzZXRDb2RlLCBGdW5jdGlvbiwgUnVudGltZSB9IGZyb20gJ0Bhd3MtY2RrL2F3cy1sYW1iZGEnO1xuaW1wb3J0IHsgQXBwLCBTdGFjayB9IGZyb20gJ0Bhd3MtY2RrL2NvcmUnO1xuaW1wb3J0IHsgVXNlclBvb2wgfSBmcm9tICdAYXdzLWNkay9hd3MtY29nbml0bydcblxuZXhwb3J0IGNsYXNzIEF1dGhTdGFjayBleHRlbmRzIFN0YWNrIHtcbiAgY29uc3RydWN0b3IoYXBwOiBBcHAsIGlkOiBzdHJpbmcpIHtcbiAgICBzdXBlcihhcHAsIGlkKTtcblxuICAgIC8vIEZ1bmN0aW9uIHRoYXQgcmV0dXJucyAyMDEgd2l0aCBcIkhlbGxvIHdvcmxkIVwiXG4gICAgY29uc3QgaGVsbG9Xb3JsZEZ1bmN0aW9uID0gbmV3IEZ1bmN0aW9uKHRoaXMsICdoZWxsb1dvcmxkRnVuY3Rpb24nLCB7XG4gICAgICBjb2RlOiBuZXcgQXNzZXRDb2RlKCdsYW1iZGFzJyksXG4gICAgICBoYW5kbGVyOiAnaGVsbG93b3JsZC5oYW5kbGVyJyxcbiAgICAgIHJ1bnRpbWU6IFJ1bnRpbWUuTk9ERUpTXzEyX1hcbiAgICB9KTtcblxuICAgIC8vIFJlc3QgQVBJIGJhY2tlZCBieSB0aGUgaGVsbG9Xb3JsZEZ1bmN0aW9uXG4gICAgY29uc3QgaGVsbG9Xb3JsZExhbWJkYVJlc3RBcGkgPSBuZXcgTGFtYmRhUmVzdEFwaSh0aGlzLCAnaGVsbG9Xb3JsZExhbWJkYVJlc3RBcGknLCB7XG4gICAgICByZXN0QXBpTmFtZTogJ0hlbGxvIFdvcmxkIEFQSScsXG4gICAgICBoYW5kbGVyOiBoZWxsb1dvcmxkRnVuY3Rpb24sXG4gICAgICBwcm94eTogZmFsc2UsXG4gICAgfSk7XG5cbiAgICAvLyBDb2duaXRvIFVzZXIgUG9vbCB3aXRoIEVtYWlsIFNpZ24taW4gVHlwZS5cbiAgICBjb25zdCB1c2VyUG9vbCA9IG5ldyBVc2VyUG9vbCh0aGlzLCAndXNlclBvb2wnLCB7XG4gICAgICBzaWduSW5BbGlhc2VzOiB7XG4gICAgICAgIGVtYWlsOiB0cnVlXG4gICAgICB9XG4gICAgfSlcblxuICAgIC8vIEF1dGhvcml6ZXIgZm9yIHRoZSBIZWxsbyBXb3JsZCBBUEkgdGhhdCB1c2VzIHRoZVxuICAgIC8vIENvZ25pdG8gVXNlciBwb29sIHRvIEF1dGhvcml6ZSB1c2Vycy5cbiAgICBjb25zdCBhdXRob3JpemVyID0gbmV3IENmbkF1dGhvcml6ZXIodGhpcywgJ2NmbkF1dGgnLCB7XG4gICAgICByZXN0QXBpSWQ6IGhlbGxvV29ybGRMYW1iZGFSZXN0QXBpLnJlc3RBcGlJZCxcbiAgICAgIG5hbWU6ICdIZWxsb1dvcmxkQVBJQXV0aG9yaXplcicsXG4gICAgICB0eXBlOiAnQ09HTklUT19VU0VSX1BPT0xTJyxcbiAgICAgIGlkZW50aXR5U291cmNlOiAnbWV0aG9kLnJlcXVlc3QuaGVhZGVyLkF1dGhvcml6YXRpb24nLFxuICAgICAgcHJvdmlkZXJBcm5zOiBbdXNlclBvb2wudXNlclBvb2xBcm5dLFxuICAgIH0pXG5cbiAgICAvLyBIZWxsbyBSZXNvdXJjZSBBUEkgZm9yIHRoZSBSRVNUIEFQSS4gXG4gICAgY29uc3QgaGVsbG8gPSBoZWxsb1dvcmxkTGFtYmRhUmVzdEFwaS5yb290LmFkZFJlc291cmNlKCdIRUxMTycpO1xuXG4gICAgLy8gR0VUIG1ldGhvZCBmb3IgdGhlIEhFTExPIEFQSSByZXNvdXJjZS4gSXQgdXNlcyBDb2duaXRvIGZvclxuICAgIC8vIGF1dGhvcml6YXRpb24gYW5kIHRoZSBhdWF0aG9yaXplciBkZWZpbmVkIGFib3ZlLlxuICAgIGhlbGxvLmFkZE1ldGhvZCgnR0VUJywgbmV3IExhbWJkYUludGVncmF0aW9uKGhlbGxvV29ybGRGdW5jdGlvbiksIHtcbiAgICAgIGF1dGhvcml6YXRpb25UeXBlOiBBdXRob3JpemF0aW9uVHlwZS5DT0dOSVRPLFxuICAgICAgYXV0aG9yaXplcjoge1xuICAgICAgICBhdXRob3JpemVySWQ6IGF1dGhvcml6ZXIucmVmXG4gICAgICB9XG4gICAgICBcbiAgICB9KVxuICAgIFxuICB9XG59XG5cblxuY29uc3QgYXBwID0gbmV3IEFwcCgpO1xubmV3IEF1dGhTdGFjayhhcHAsICdBdXRoU3RhY2snKTtcbmFwcC5zeW50aCgpOyJdfQ==