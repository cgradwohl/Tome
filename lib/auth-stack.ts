import { LambdaRestApi, CfnAuthorizer, LambdaIntegration, AuthorizationType } from '@aws-cdk/aws-apigateway';
import { AssetCode, Function, Runtime } from '@aws-cdk/aws-lambda';
import { App, Stack, CfnOutput } from '@aws-cdk/core';
import { UserPool, UserPoolClient } from '@aws-cdk/aws-cognito'

export class AuthStack extends Stack {
  constructor(app: App, id: string) {
    super(app, id);

    // Cognito User Pool with Email Sign-in Type.
    const userPool = new UserPool(this, 'TomeUserPool', {
      signInAliases: {
        email: true
      }
    })

    const userPoolClient = new UserPoolClient(this, 'TomeUserPoolClient', {
      userPoolClientName: 'testClient',
      userPool: userPool,
      authFlows: {
        adminUserPassword: true,
        refreshToken: true,
      },
      preventUserExistenceErrors: true
    });

    // Function that returns 201 with "Hello world!"
    const helloWorldFunction = new Function(this, 'helloWorldFunction', {
      code: new AssetCode('lambdas'),
      handler: 'helloworld.handler',
      runtime: Runtime.NODEJS_12_X,
    });


    // Rest API backed by the helloWorldFunction
    // Defines an API Gateway REST API with AWS Lambda proxy integration
    const helloWorldLambdaRestApi = new LambdaRestApi(this, 'helloWorldLambdaRestApi', {
      restApiName: 'Hello World API',
      handler: helloWorldFunction,
      proxy: false,
    });

    // Authorizer for the Hello World API that uses the
    // Cognito User pool to Authorize users.
    // AWS::ApiGateway::Authorizer
    const authorizer = new CfnAuthorizer(this, 'cfnAuth', {
      restApiId: helloWorldLambdaRestApi.restApiId,
      name: 'HelloWorldAPIAuthorizer',
      type: 'COGNITO_USER_POOLS',
      identitySource: 'method.request.header.Authorization',
      providerArns: [userPool.userPoolArn],
    })

    // Hello Resource API for the REST API. 
    const hello = helloWorldLambdaRestApi.root.addResource('HELLO');

    // GET method for the HELLO API resource. It uses Cognito for
    // authorization and the auathorizer defined above.
    hello.addMethod('GET', new LambdaIntegration(helloWorldFunction), {
      authorizationType: AuthorizationType.COGNITO,
      authorizer: {
        authorizerId: authorizer.ref
      }
    })

    new CfnOutput(this, "UserPoolIdOutput", {
      description: "UserPool ID",
      value: userPool.userPoolId
    });
    
  }
}


const app = new App();
new AuthStack(app, 'AuthStack');
app.synth();