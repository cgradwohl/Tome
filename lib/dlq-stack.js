"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cloudwatch = require("@aws-cdk/aws-cloudwatch");
const sqs = require("@aws-cdk/aws-sqs");
class DeadLetterQueue extends sqs.Queue {
    constructor(scope, id) {
        super(scope, id);
        // Add the alarm
        this.messagesInQueueAlarm = new cloudwatch.Alarm(this, 'Alarm', {
            alarmDescription: 'There are messages in the Dead Letter Queue',
            evaluationPeriods: 1,
            threshold: 1,
            metric: this.metricApproximateNumberOfMessagesVisible(),
        });
    }
}
exports.DeadLetterQueue = DeadLetterQueue;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiZGxxLXN0YWNrLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiZGxxLXN0YWNrLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7O0FBQUEsc0RBQXNEO0FBQ3RELHdDQUF3QztBQUd4QyxNQUFhLGVBQWdCLFNBQVEsR0FBRyxDQUFDLEtBQUs7SUFHNUMsWUFBWSxLQUFnQixFQUFFLEVBQVU7UUFDdEMsS0FBSyxDQUFDLEtBQUssRUFBRSxFQUFFLENBQUMsQ0FBQztRQUVqQixnQkFBZ0I7UUFDaEIsSUFBSSxDQUFDLG9CQUFvQixHQUFHLElBQUksVUFBVSxDQUFDLEtBQUssQ0FBQyxJQUFJLEVBQUUsT0FBTyxFQUFFO1lBQzlELGdCQUFnQixFQUFFLDZDQUE2QztZQUMvRCxpQkFBaUIsRUFBRSxDQUFDO1lBQ3BCLFNBQVMsRUFBRSxDQUFDO1lBQ1osTUFBTSxFQUFFLElBQUksQ0FBQyx3Q0FBd0MsRUFBRTtTQUN4RCxDQUFDLENBQUM7SUFDTCxDQUFDO0NBQ0Y7QUFkRCwwQ0FjQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAqIGFzIGNsb3Vkd2F0Y2ggZnJvbSAnQGF3cy1jZGsvYXdzLWNsb3Vkd2F0Y2gnO1xuaW1wb3J0ICogYXMgc3FzIGZyb20gJ0Bhd3MtY2RrL2F3cy1zcXMnO1xuaW1wb3J0IHsgQ29uc3RydWN0LCBEdXJhdGlvbiB9IGZyb20gJ0Bhd3MtY2RrL2NvcmUnO1xuXG5leHBvcnQgY2xhc3MgRGVhZExldHRlclF1ZXVlIGV4dGVuZHMgc3FzLlF1ZXVlIHtcbiAgcHVibGljIHJlYWRvbmx5IG1lc3NhZ2VzSW5RdWV1ZUFsYXJtOiBjbG91ZHdhdGNoLklBbGFybTtcblxuICBjb25zdHJ1Y3RvcihzY29wZTogQ29uc3RydWN0LCBpZDogc3RyaW5nKSB7XG4gICAgc3VwZXIoc2NvcGUsIGlkKTtcblxuICAgIC8vIEFkZCB0aGUgYWxhcm1cbiAgICB0aGlzLm1lc3NhZ2VzSW5RdWV1ZUFsYXJtID0gbmV3IGNsb3Vkd2F0Y2guQWxhcm0odGhpcywgJ0FsYXJtJywge1xuICAgICAgYWxhcm1EZXNjcmlwdGlvbjogJ1RoZXJlIGFyZSBtZXNzYWdlcyBpbiB0aGUgRGVhZCBMZXR0ZXIgUXVldWUnLFxuICAgICAgZXZhbHVhdGlvblBlcmlvZHM6IDEsXG4gICAgICB0aHJlc2hvbGQ6IDEsXG4gICAgICBtZXRyaWM6IHRoaXMubWV0cmljQXBwcm94aW1hdGVOdW1iZXJPZk1lc3NhZ2VzVmlzaWJsZSgpLFxuICAgIH0pO1xuICB9XG59Il19