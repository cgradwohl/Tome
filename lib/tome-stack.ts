import * as cdk from '@aws-cdk/core';
import * as cognito from '@aws-cdk/aws-cognito';
export class TomeStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here

    // Auth:
    // Cognito User Pool
    // note this is from the construct library and not a CFN Construct
    const userPool = new cognito.UserPool(this, 'TomeUserPool', {
      signInAliases: {
        email: true,
      },
      signInCaseSensitive: false,
      autoVerify: {
        email: true,
      },
      passwordPolicy: {
        minLength: 8,
        requireLowercase: true,
        requireDigits: true,
        requireUppercase: true,
        requireSymbols: true
      },
      standardAttributes: {
        givenName: {
          mutable: true,
          required: true
        },
        familyName: {
          mutable: true,
          required: true
        },
        email: {
          mutable: true,
          required: true
        }
      }
    });


    // Lambda
    // DynamoDB Table
    

    // User Data: Profile, Library, Group Meta, Queue, Tome
    // API Gateway
    // Cognito Authorizer
    // Lambda w/ read access to DDD
    // DynamoDB Table

    // TODO: folder structure (infra vs application code .... or are they the same?   :))
  }
}
