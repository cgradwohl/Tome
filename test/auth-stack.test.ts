import { SynthUtils } from '@aws-cdk/assert';
import { App } from '@aws-cdk/core';

import { AuthStack } from '../lib/auth-stack';

describe('auth-stack', () => {
  it('should work', () => {
    const app = new App();

    const authStack = new AuthStack(app, 'TestAuthStack');

    expect(SynthUtils.toCloudFormation(authStack)).toMatchSnapshot();
  });
});