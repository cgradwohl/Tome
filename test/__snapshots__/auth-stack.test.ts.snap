// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth-stack should work 1`] = `
Object {
  "Outputs": Object {
    "helloWorldLambdaRestApiEndpointC5601FAC": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "helloWorldLambdaRestApi6825FB98",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "helloWorldLambdaRestApiDeploymentStageprod67DD79AF",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8ArtifactHash3A1D95D0": Object {
      "Description": "Artifact hash for asset \\"1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8\\"",
      "Type": "String",
    },
    "AssetParameters1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8S3Bucket5C95B393": Object {
      "Description": "S3 bucket for asset \\"1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8\\"",
      "Type": "String",
    },
    "AssetParameters1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8S3VersionKey84FB9C63": Object {
      "Description": "S3 key for asset version \\"1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "cfnAuth": Object {
      "Properties": Object {
        "IdentitySource": "method.request.header.Authorization",
        "Name": "HelloWorldAPIAuthorizer",
        "ProviderARNs": Array [
          Object {
            "Fn::GetAtt": Array [
              "userPoolDC9497E0",
              "Arn",
            ],
          },
        ],
        "RestApiId": Object {
          "Ref": "helloWorldLambdaRestApi6825FB98",
        },
        "Type": "COGNITO_USER_POOLS",
      },
      "Type": "AWS::ApiGateway::Authorizer",
    },
    "helloWorldFunction00C940B5": Object {
      "DependsOn": Array [
        "helloWorldFunctionServiceRole8475DBF0",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8S3Bucket5C95B393",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8S3VersionKey84FB9C63",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1b13b8775392bc0532db830d8447779bacfe88e0b36f20766ebf1e2085091db8S3VersionKey84FB9C63",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "helloworld.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "helloWorldFunctionServiceRole8475DBF0",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "helloWorldFunctionServiceRole8475DBF0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "helloWorldLambdaRestApi6825FB98": Object {
      "Properties": Object {
        "Name": "Hello World API",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "helloWorldLambdaRestApiAccount3D66EADA": Object {
      "DependsOn": Array [
        "helloWorldLambdaRestApi6825FB98",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "helloWorldLambdaRestApiCloudWatchRole22367FBD",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "helloWorldLambdaRestApiCloudWatchRole22367FBD": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "helloWorldLambdaRestApiDeploymentCB1FF574ae10c162afd462d14276188f96f29d56": Object {
      "DependsOn": Array [
        "helloWorldLambdaRestApiHELLOGET6E88F46C",
        "helloWorldLambdaRestApiHELLOAD091B67",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "helloWorldLambdaRestApi6825FB98",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "helloWorldLambdaRestApiDeploymentStageprod67DD79AF": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "helloWorldLambdaRestApiDeploymentCB1FF574ae10c162afd462d14276188f96f29d56",
        },
        "RestApiId": Object {
          "Ref": "helloWorldLambdaRestApi6825FB98",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "helloWorldLambdaRestApiHELLOAD091B67": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "helloWorldLambdaRestApi6825FB98",
            "RootResourceId",
          ],
        },
        "PathPart": "HELLO",
        "RestApiId": Object {
          "Ref": "helloWorldLambdaRestApi6825FB98",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "helloWorldLambdaRestApiHELLOGET6E88F46C": Object {
      "Properties": Object {
        "AuthorizationType": "COGNITO_USER_POOLS",
        "AuthorizerId": Object {
          "Ref": "cfnAuth",
        },
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "helloWorldFunction00C940B5",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "helloWorldLambdaRestApiHELLOAD091B67",
        },
        "RestApiId": Object {
          "Ref": "helloWorldLambdaRestApi6825FB98",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "helloWorldLambdaRestApiHELLOGETApiPermissionTestAuthStackhelloWorldLambdaRestApiDD8D8853GETHELLOEDFC2CE9": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "helloWorldFunction00C940B5",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "helloWorldLambdaRestApi6825FB98",
              },
              "/",
              Object {
                "Ref": "helloWorldLambdaRestApiDeploymentStageprod67DD79AF",
              },
              "/GET/HELLO",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "helloWorldLambdaRestApiHELLOGETApiPermissionTestTestAuthStackhelloWorldLambdaRestApiDD8D8853GETHELLO6A5EDC0F": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "helloWorldFunction00C940B5",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "helloWorldLambdaRestApi6825FB98",
              },
              "/test-invoke-stage/GET/HELLO",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "userPoolDC9497E0": Object {
      "Properties": Object {
        "AccountRecoverySetting": Object {
          "RecoveryMechanisms": Array [
            Object {
              "Name": "verified_phone_number",
              "Priority": 1,
            },
            Object {
              "Name": "verified_email",
              "Priority": 2,
            },
          ],
        },
        "AdminCreateUserConfig": Object {
          "AllowAdminCreateUserOnly": true,
        },
        "AutoVerifiedAttributes": Array [
          "email",
        ],
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "SmsConfiguration": Object {
          "ExternalId": "TestAuthStackuserPool1870016B",
          "SnsCallerArn": Object {
            "Fn::GetAtt": Array [
              "userPoolsmsRoleED3E1FD9",
              "Arn",
            ],
          },
        },
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UsernameAttributes": Array [
          "email",
        ],
        "VerificationMessageTemplate": Object {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
    },
    "userPoolsmsRoleED3E1FD9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Condition": Object {
                "StringEquals": Object {
                  "sts:ExternalId": "TestAuthStackuserPool1870016B",
                },
              },
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cognito-idp.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "sns:Publish",
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "sns-publish",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
